src/main/java/graphql/execution/NonNullableFieldValidator.java

```diff
@@ -46,8 +47,15 @@ public <T> T validate(ResultPath path, T result) throws NonNullableFieldWasNullE
                 //
                 // We will do this until the spec makes this more explicit.
                 //
+                final ResultPath path = parameters.getPath();
+
                 NonNullableFieldWasNullException nonNullException = new NonNullableFieldWasNullException(executionStepInfo, path);
-                executionContext.addError(new NonNullableFieldWasNullError(nonNullException), path);
+                final GraphQLError error = new NonNullableFieldWasNullError(nonNullException);
+                if(parameters.getDeferredCallContext() != null) {
+                    parameters.getDeferredCallContext().addError(error);
+                } else {
+                    executionContext.addError(error, path);
+                }
```

Comment:
I wonder if there's a better way to do this, because this is fairly error prone.

i.e. if you're adding an error you have to remember to check for the presence of a deferred context first.

Could we clone the `ExecutionContext` for a defer and have a separate error array?

ofc this is only a problem for framework code, and end users should be ok… unless they invoke `ExecutionContext.addError` themselves? Seems like it's public API?

Reply:
I totally agree about it being high error-prone, but I couldn't figure out a simple solution. As long as `ExecutionContext#addError` exists, any code can always add errors directly to `ExecutionContext` without checking for defer...

> Could we clone the ExecutionContext for a defer and have a separate error array?
hm... `DeferredCallContext` is sort of that. Is a context but only with the state that is relevant for a defer block, which for now is just a list of errors.
Can you expand on your suggestion? :-)